dimod/binary_quadratic_model/model.py

```diff
@@ -0,0 +1,1099 @@
+
+        Args:
+            interactions (iterable[[variable, variable]]):
+                A collections of interactions. Each interaction should be a 2-tuple of variables
+                in the binary quadratic model.
+
+        Notes:
+            Any interaction not in the binary quadratic model is ignored.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel({}, {('a', 'b'): -1, ('b', 'c'): 1}, 0.0, dimod.SPIN)
+            >>> bqm.remove_interactions_from([('b', 'c'), ('a', 'c')])  # ('a', 'c') is not an interaction, so ignored
+            >>> bqm.quadratic
+            {('a', 'b'): -1}
+
+        """
+        for u, v in interactions:
+            self.remove_interaction(u, v)
+
+    def add_offset(self, offset):
+        """Add given value to the offset.
+
+        Args:
+            offset (number):
+                A value to be added to the constant energy offset for the binary quadratic model.
+
+        """
+        self.offset += offset
+
+        try:
+            self._counterpart.add_offset(offset)
+        except AttributeError:
+            pass
+
+    def remove_offset(self):
+        """Set the binary quadratic model's offset to zero.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel({}, {}, 1.3, dimod.SPIN)
+            >>> bqm.remove_offset()
+            >>> bqm.offset
+            0.0
+
+        """
+        self.add_offset(-self.offset)
+
+    def scale(self, scalar):
+        """Multiply all of the biases and the offset by the given scalar.
+
+        Args:
+            scalar (number):
+                The value to scale the energy range of the binary quadratic model by.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel({'a': -2, 'b': 2}, {('a', 'b'): -1}, 1., dimod.SPIN)
+            >>> bqm.scale(.5)
+            >>> bqm.linear
+            {'a': -1., 'b': 1.}
+            >>> bqm.quadratic
+            {('a', 'b'): -.5}
+            >>> bqm.offset
+            .5
+
+        """
+        if not isinstance(scalar, Number):
+            raise TypeError("expected scalar to be a Number")
+
+        linear = self.linear
+        for v in linear:
+            linear[v] *= scalar
+
+        quadratic = self.quadratic
+        for edge in quadratic:
+            quadratic[edge] *= scalar
+
+        adj = self.adj
+        for u in adj:
+            for v in adj[u]:
+                adj[u][v] *= scalar
+
+        self.offset *= scalar
+
+        try:
+            self._counterpart.scale(scalar)
+        except AttributeError:
+            pass
+
+    def fix_variable(self, v, value):
+        """Fix the value of a variable in the binary quadratic model and remove it.
+
+        Args:
+            v (variable):
+                A variable in the binary quadratic model that has an interaction with u.
+
+            value (int):
+                The value assigned to the variable, must match the :class:`.Vartype` of the binary
```

Comment:
Could be nice if this were flexible to be either `.Vartype` as with the above functions.

Reply:
Doesn't really make sense in this context, also want to force the user to be deliberate about it.